### PR TITLE
[Feature-9772][plugin/ui] spark sql support for SparkTask

### PR DESCRIPTION
**Pull request by sq-q**
## Purpose of the pull request

According to user requirements, we hope to support spark sql for SparkTask.

Users want to have an editor to fill in the sql script, which is convenient for ETL.

The renderings we need are as follows:
lQLPDhtd5ypCZK3NA5bNBCywAi-2JJ5_quwCaoYbSoA-AA_1068_918

## Brief change log

- ProgramType.class add sql type
- SparkTask.class and SparkArgsUtils.class add main logical for support Spark sql command
- use-spark.ts adds some logic to support front-end pages supporting spark sql editing SQL

## Verify this pull request

This change added tests and can be verified as follows:

org.apache.dolphinscheduler.plugin.task.spark.SparkTaskTest

### CODE REVIEW

File: dolphinscheduler-common/src/main/resources/common.properties [hunk 1]
```diff
@@ -92,4 +92,4 @@ development.state=false
 alert.rpc.port=50052
 
 # Url endpoint for zeppelin RESTful API
-zeppelin.rest.url="http://localhost:8080"
+zeppelin.rest.url="http://localhost:8080"
```
Comment (SbloodyS) [inline dolphinscheduler-common/src/main/resources/common.properties]: We should avoid removing the last line.

File: dolphinscheduler-master/src/main/resources/logback-spring.xml [hunk 2]
```diff
@@ -72,4 +72,4 @@
         <appender-ref ref="TASKLOGFILE"/>
         <appender-ref ref="MASTERLOGFILE"/>
     </root>
-</configuration>
+</configuration>
```
Comment (SbloodyS) [inline dolphinscheduler-master/src/main/resources/logback-spring.xml]: Same here.

File: dolphinscheduler-ui-next/src/views/projects/task/components/node/fields/use-spark.ts [hunk 3]
```diff
@@ -136,4 +164,4 @@ export const SPARK_VERSIONS = [
     label: 'SPARK1',
     value: 'SPARK1'
   }
-]
+]
```
Comment (SbloodyS) [inline dolphinscheduler-ui-next/src/views/projects/task/components/node/fields/use-spark.ts]: Same here.

File: dolphinscheduler-tools/src/main/resources/application.yaml [hunk 4]
```diff
@@ -43,4 +43,4 @@ spring:
       on-profile: mysql
   datasource:
     driver-class-name: com.mysql.jdbc.Driver
-    url: jdbc:mysql://127.0.0.1:3306/dolphinscheduler?useUnicode=true&characterEncoding=UTF-8
+    url: jdbc:mysql://localhost/dolphinscheduler?useUnicode=true&characterEncoding=UTF-8
```
Comment (SbloodyS) [inline dolphinscheduler-tools/src/main/resources/application.yaml]: Why did you change this?

**Comment by SbloodyS** [comment]
Hi @sq-q , please check failed CI.

File: dolphinscheduler-common/src/main/resources/common.properties (hunk 1 above)
Comment (sq-q) [reply dolphinscheduler-common/src/main/resources/common.properties]: I am changing

File: dolphinscheduler-tools/src/main/resources/application.yaml (hunk 4 above)
Comment (sq-q) [reply dolphinscheduler-tools/src/main/resources/application.yaml]: I am modifying

**Comment by sq-q** [comment]
> Hi @sq-q , please check failed CI.
